test/goose/brokers/redis/console/page_test.clj

```diff
@@ -85,8 +79,90 @@
       (true? (spy/called-once? enqueued/get-jobs))
       (is (= [{:status 200
                :body   "Mocked resp"}] (spy/responses enqueued/get-jobs)))
-      (is (= "default" (get-in (first (spy/first-call enqueued/get-jobs)) [:params :queue]))))
+      (is (= "default" (get-in (first (spy/first-call enqueued/get-jobs)) [:params :queue])))))
+
+  (testing "Main handler should invoke purge-queue handler"
     (with-redefs [enqueued/purge-queue (spy/stub {:status 302 :headers {"Location" "/enqueued"} :body ""})]
       (console/handler tu/redis-producer (mock/request :delete "/enqueued/queue/default"))
       (true? (spy/called-once? enqueued/purge-queue))
-      (is (= [{:status 302 :headers {"Location" "/enqueued"} :body ""}] (spy/responses enqueued/purge-queue))))))
+      (is (= [{:status 302 :headers {"Location" "/enqueued"} :body ""}] (spy/responses enqueued/purge-queue)))))
+
+  (testing "Main handler should invoke delete-jobs handler"
+    (with-redefs [enqueued/delete-jobs (spy/stub {:status 302 :headers {"Location" "/enqueued/queue/test"} :body ""})]
+      (console/handler tu/redis-producer (mock/request :delete "/enqueued/queue/default/jobs"))
+      (true? (spy/called-once? enqueued/delete-jobs))
+      (is (= [{:status 302 :headers {"Location" "/enqueued/queue/test"} :body ""}] (spy/responses enqueued/delete-jobs)))))
+
+  (testing "Main handler should invoke prioritise-jobs handler"
+    (with-redefs [enqueued/prioritise-jobs (spy/stub {:status 302 :headers {"Location" "/enqueued/queue/test"} :body ""})]
+      (console/handler tu/redis-producer (mock/request :post "/enqueued/queue/default/jobs"))
+      (true? (spy/called-once? enqueued/prioritise-jobs))
+      (is (= [{:status 302 :headers {"Location" "/enqueued/queue/test"} :body ""}] (spy/responses enqueued/prioritise-jobs))))))
+
+(deftest purge-queue-test
+  (testing "Should purge a queue"
+    (f/create-async-job)
+    (f/create-async-job {:queue       "queue1"
+                         :ready-queue "goose/queue:queue1"})
+    (is (true? (every? #{tu/queue "queue1"} (enqueued-jobs/list-all-queues tu/redis-conn))))
+    (is (= 2 (count (enqueued-jobs/list-all-queues tu/redis-conn))))
+    (is (= {:body    ""
+            :headers {"Location" "/enqueued"}
+            :status  302} (enqueued/purge-queue {:console-opts tu/redis-console-opts
+                                                 :params       {:queue tu/queue}
+                                                 :prefix-route str})))
+    (is (= ["queue1"] (enqueued-jobs/list-all-queues tu/redis-conn)))))
+
+(deftest delete-jobs-test
```

Comment:
question(non-blocking): do we silently fail if the job has already been executed i.e no longer in the queue?

Reply:
- Yes, we silently fail if the job is no longer in the queue. I will keep a note on adding it the docs in the release.
- I think showing an error might create a lot of noise in the UI if multiple job deletion is unsuccessful. Additionally there isn't any action(unlike job creation failure where we create a job again) to be taken if a job to be deleted is already executed.